BlockEntity/BECharcoalPit.cs

```diff
@@ -255,35 +252,70 @@ BlockPos FindHoleInPit()
+                    bool solid = nBlock.GetLiquidBarrierHeightOnSide(facing.Opposite, npos) == 1 || nBlock.GetLiquidBarrierHeightOnSide(facing, bpos) == 1;
                     bool isFirewoodpile = BlockFirepit.IsFirewoodPile(Api.World, npos);
 
-                    if (!solid && !isFirewoodpile && nBlock.BlockId != charcoalPitBlockId)
+                    if (!isFirewoodpile && nBlock.BlockId != charcoalPitBlockId)
                     {
-                        return bpos;
+                        if (IsCombustible(npos)) holes.Add(npos.Copy());
+                        else if (!solid) holes.Add(bpos.Copy());
                     }
 
                     // Only traverse inside the firewood pile
                     if (!isFirewoodpile) continue;
 
-                    // Only traverse within a 12x12x12 block cube
-                    bool inCube = Math.Abs(npos.X - Pos.X) <= maxHalfSize && Math.Abs(npos.Y - Pos.Y) <= maxHalfSize && Math.Abs(npos.Z - Pos.Z) <= maxHalfSize;
-                    
-                    if (inCube && !visitedPositions.Contains(npos))
+                    if (InCube(npos, ref minPos, ref maxPos) && !visitedPositions.Contains(npos))
                     {
-                        bfsQueue.Enqueue(npos);
-                        visitedPositions.Add(npos);
+                        bfsQueue.Enqueue(npos.Copy());
+                        visitedPositions.Add(npos.Copy());
                     }
                 }
             }
 
-            return null;
+            return holes;
+        }
+
+        private bool InCube(BlockPos npos, ref BlockPos minPos, ref BlockPos maxPos)
+        {
+            BlockPos nmin = minPos.Copy(), nmax = maxPos.Copy();
+
+            if (npos.X < minPos.X) nmin.X = npos.X;
+            else if (npos.X > maxPos.X) nmax.X = npos.X;
+
+            if (npos.Y < minPos.Y) nmin.Y = npos.Y;
+            else if (npos.Y > maxPos.Y) nmax.Y = npos.Y;
+
+            if (npos.Z < minPos.Z) nmin.Z = npos.Z;
+            else if (npos.Z > maxPos.Z) nmax.Z = npos.Z;
+
+            // Only traverse within maxSize range
+            if (nmax.X - nmin.X + 1 <= maxSize && nmax.Y - nmin.Y + 1 <= maxSize && nmax.Z - nmin.Z + 1 <= maxSize)
+            {
+                minPos = nmin.Copy();
+                maxPos = nmax.Copy();
+                return true;
+            }
+
+            return false;
+        }
+
+        private bool IsCombustible(BlockPos pos)
+        {
+            Block block = Api.World.BlockAccessor.GetBlock(pos);
+            if (block.CombustibleProps != null) return block.CombustibleProps.BurnDuration > 0;
+
+            if (block is ICombustible bic)
+            {
+                return bic.GetBurnDuration(Api.World, pos) > 0;
+            }
+
+            return false;

```

Comment:
This function is just used by FindHolesInPit to check whether or not the walls that make up the pit are flammable.  That way even though something like a solid oak door _would_ count as a solid wall, it will light on fire and still not work effectively for the charcoal pit, necessitating that all walls of the pit be built out of properly fire resistant materials.